python/pyarrow/types.pxi

```diff
@@ -1278,6 +1278,46 @@ cdef class Schema:
         return self.sp_schema.get().Equals(deref(other.schema),
                                            check_metadata)
 
+    @classmethod
+    def merge(cls, list schemas):
```

Comment:
Alternative could also be a `unify_schemas` *function* (as https://github.com/apache/arrow/pull/6890), instead of a class method (eg concat_table is also a function and not a classmethod)